### PR TITLE
test: Missing E2E tests for B2B/B2C Checkout Flow (#2478)

### LINKED ISSUE #2478: [still needed?] Create new e2e test to check extensible checkout improvements
Create new e2e tests for extensible checkout.
- [ ] Test if guards correctly protect routes.
- [ ] Check if url is correctly set on each step
- [ ] any other important test you might think of

### PR DESCRIPTION
**Pull request by pawelfras**
closes #2478

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts [hunk 1]
```diff
@@ -336,17 +337,35 @@ export function addCheapProductToCart(
...
+  cy.get('cx-order-summary .cx-summary-partials .cx-summary-total')
     .find('.cx-summary-amount')
     .should('contain', cartData.total);
+}
+
+export function fillAddressFormWithCheapProduct(
+  shippingAddressData: Partial<AddressData> = user
+) {
+  cy.log('🛒 Filling shipping address form');
+  cy.log(JSON.stringify(shippingAddressData));
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts]: remove comment

**Review by dydome — CHANGES_REQUESTED** [review]

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts [hunk 2]
```diff
@@ -1,3 +1,10 @@
+//TODO: find better place for this type
+export type DeepPartial<T> = T extends object
```
Comment (szczepaniak-michal) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts]: let's move it to separate file under helpers dir

Comment (pawelfras) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts]: Type has been moved to `form.ts` file under helpers dir

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts (hunk 1 above)
Comment (pawelfras) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts]: removed 👍

**Review by dydome — APPROVED** [review]

**Review by szczepaniak-michal — APPROVED** [review]

**Pull request merged**